### PR TITLE
Add CMAC to the benchmark example

### PR DESCRIPTION
**Pull request by andresag01**
This PR introduces code to the benchmark AES-CMAC and 3DES-CMAC.

I have made this PR dependent on https://github.com/ARMmbed/oob-mbed-os-example-tls/pull/11 to avoid merge conflicts.

@sbutcher-arm

### CODE REVIEW

File: benchmark/main.cpp [hunk 1]
```diff
@@ -450,6 +462,22 @@ static int benchmark( int argc, char *argv[] )
...
+        unsigned char output[8];
+        const mbedtls_cipher_info_t *cipher_info;
+
+        memset( buf, 0, sizeof( buf ) );
+        memset( tmp, 0, sizeof( tmp ) );
+
+        cipher_info = mbedtls_cipher_info_from_type( MBEDTLS_CIPHER_DES_EDE3_ECB );
+
+        TIME_AND_TSC( "3DES-CMAC",
+                      mbedtls_cipher_cmac( cipher_info, tmp, 192, buf,
```
Comment (yanesca) [inline benchmark/main.cpp]: It would be better to use `cipher_info->key_bitlen`, instead of 192.

File: benchmark/main.cpp [hunk 2]
```diff
@@ -450,6 +462,22 @@ static int benchmark( int argc, char *argv[] )
                 mbedtls_des_crypt_cbc( &des, MBEDTLS_DES_ENCRYPT, BUFSIZE, tmp, buf, buf ) );
         mbedtls_des_free( &des );
     }
+#if defined(MBEDTLS_CMAC_C)
+    if( todo.des3_cmac )
+    {
+        unsigned char output[8];
```
Comment (yanesca) [inline benchmark/main.cpp]: I would prefer `MBEDTLS_CIPHER_BLKSIZE_MAX` here.

File: benchmark/main.cpp [hunk 3]
```diff
@@ -519,6 +547,37 @@ static int benchmark( int argc, char *argv[] )
         }
     }
 #endif
+#if defined(MBEDTLS_CMAC_C)
+    if( todo.aes_cmac )
+    {
+        unsigned char output[16];
```
Comment (yanesca) [inline benchmark/main.cpp]: I would prefer `MBEDTLS_CIPHER_BLKSIZE_MAX` here.

**Comment by andresag01** [comment]
Rebased changes on top of master

**Comment by andresag01** [comment]
Reviewed and approved. Merging...

**Pull request merged**